asteroid/globals.py

```diff
@@ -167,3 +167,20 @@ def __str__(self):
     'struct',
     'typematch',
 ]
+
+###########################################################################################
+# list of structures that a pattern type-pattern will subsume/overlap
+
+pattern_subsumes = [
```

Comment:
Your `pattern_subsumes` and the `patterns` list only differ by one element.

I would make `patterns` a set and then define `pattern_subsumes` as,
```
pattern_subsumes = patterns - {'id'}
```
that way we avoid repeated definitions of almost the same things in the code.

Reply:
This should be addressed and updated now.